rfcs/react-components/no-action-prop-value-standardization.md

```diff
@@ -0,0 +1,97 @@
+# RFC: "No action" / default property value standardization
+
+---
+
+_@andrefcdias @Hotell_
+
+## Summary
+
+This RFC aims to standardize what values we use in our components for cases where a prop value results in no action, i.e. defaults that apply no styles.
+
+## Background
+
+Currently, the approach we follow for all components is to use a string value like 'off', 'none' or 'default' for default values of a prop.
+
+## Problem statement
+
+There is no standardization for the naming, resulting in our users needing to read documentation to figure out what to use and requiring specific component knowledge as they can't reuse this information for other components. Given that we use TypeScript's `Required` to enforce these attributes to be non-nullable, a user will have to read through all the docs of said component to figure out what to use, when consuming a `useComponentStyles_unstable` hook. For a user that wants to consume the `useTextStyles` to apply only bold, they would need to write:
+
+```ts
+useTextStyles_unstable({
+  align: 'start',
+  block: false,
+  font: 'base',
+  italic: false,
+  size: 300,
+  strikethrough: false,
+  truncate: false,
+  underline: false,
+  weight: 'semibold', // Only change from defaults
+  wrap: true,
+});
+```
+
+## Detailed Design or Proposal
+
+This RFC proposes that we leverage the standard JavaScript default, `undefined`, for attributes instead of a string.
+For this, we should stop using `Required` for all props in the `State` and decide case by case if something should / should not be optional.
+
+Cases where our component should have a default value, we would apply the state that should be the default value.
```

Comment:
does this mean that for discriminate unions with two types we should actually convert it to a prop to a single boolean value ?

````tsx`
// before
<Button iconPosition="before" />
<Button iconPosition="after" />

//after
<Button iconAfter />
```

or do you mean that `undefined` in this case just implies that one of the options is used?

Reply:
Second option. What I mean here is that for props where the default value has side effects, a value, of any type, should be used as default instead of persisting an `undefined` to the state.
For example, with Card, we don't want a case where a user does not apply any `appearance`. As such, 'filled' is used as a default and passed into the state.
However, for example `focusMode`, when this value is `undefined`, nothing is done to the component. As such, using the underlying `undefined`, instead of defaulting to something like `"off"`, allows us to have a standard behavior across all components.

Reply:
AH ok that makes sense could you please make that more explicit (maybe with a code example) in the RFC ? it was not clear for me

Reply:
On it 👍🏻